modules/fedimint-ln/src/lib.rs

```diff
@@ -83,7 +83,8 @@ use crate::db::{
 #[derive(Debug)]
 pub struct LightningModule {
     cfg: LightningModuleConfig,
-    db: Database,
+    // TODO: remove DB all together
+    non_consensus_db: Database,
```

Comment:
What is `non_consensus_db` for?

Reply:
It's a reference to the same DB as before. I mainly renamed it to see where it's used (and change most places to use an externally supplied DB tx) and also to warn others not to use it except for operations that happen outside consensus functions.

In consensus relevant operations we want absolute consistency. E.g. if someone tries to double spend e-cash, the spend book will be updated in the database transaction (to rollback in case the entire fedimint transaction fails later because of another input/output). If we now try to validate the next spend of an e-cash token and read from the DB directly, without the DB tx applied, we won't see the updated spend book and might allow double spends. Currently `ConflictFilter` prevents that, but just reading from the DB tx is so much cleaner.

Non-consensus relevant operations are e.g. fetching an output outcome. We don't care about consistency all that much here. E.g. when fetching `OutputOutcome`s for an entire fedimint transaction some answers might be from epoch n and some from n+1. Eventually I'd like to fix that too and supply a DB tx to every operation, but for now I just want to get the modularization finished (hence the TODO).

Reply:
I also just noticed that I called it `read_only_db` in the first two commits and forgot to change that, will fix.

Reply:
Just for my understanding, when you say supply a DB tx to every operation, do you mean the same DB tx as the consensus operations or a different one? I suppose it might not matter since these read operations aren't as sensitive as the consensus operations?

Reply:
I'd use one DB tx for each atomic operation and then give it to the sub-functions. E.g. fetching the transaction status is an atomic operation and should use one DB tx to read the output outcomes from all the modules so the reply is consistent:

https://github.com/fedimint/fedimint/blob/bace3fbe9df408b1ede6b2122303ba6d2e208342/fedimint-server/src/consensus/mod.rs#L527-L591

Or processing one fedimint transaction/epoch (there it's already the case that we use the same DB tx).

Reply:
Should the non-consensus use of the DB be read-only then?

Reply:
Not necessarily, e.g. submitting a tx, while not on the module level, is a non-consensus write OP. Similar things are imaginable for e.g. backup storage in a module.